sql/core/src/test/scala/org/apache/spark/sql/execution/bucketing/CoalesceBucketsInJoinSuite.scala

```diff
@@ -178,7 +244,24 @@ class CoalesceBucketsInSortMergeJoinSuite extends SQLTestUtils with SharedSparkS
         rightKeys = rCols.reverse,
         leftRelation = lRel,
         rightRelation = RelationSetting(rCols, 8, Some(4)),
-        isSortMergeJoin = true))
+        joinOperator = SORT_MERGE_JOIN,
+        shjBuildSide = None))
+
+      run(JoinSetting(
+        leftKeys = lCols.reverse,
+        rightKeys = rCols.reverse,
+        leftRelation = lRel,
+        rightRelation = RelationSetting(rCols, 8, Some(4)),
```

Comment:
wait, I think we can't coalesce the build side?

Reply:
@cloud-fan - am I missing anything? the test is coalescing right side where we build left side.

Reply:
oh sorry I misread the code. I see a similar test below which changes the build side and I thought one must be wrong. Actually you switch the join keys.

Reply:
@cloud-fan yes, thanks for reviewing so carefully. Updated the PR and addressed comments to remove duplicated tests. Thanks.